src/mixins/MemoryMixin.h

```diff
@@ -30,7 +30,7 @@ namespace reactivesocket {
 /// implicitly specified as virtual, depending on whether the Base class
 /// is implementing the (virtual) methods of the
 /// Subscription or the Subscriber interface.
-template <typename Base>
+template <typename Base, typename Payload = Payload>
```

Comment:
If I understand the intention of your change, shouldn't this be?

`typename T = Payload`

Reactive Streams isn't tied to Payload https://github.com/ReactiveSocket/reactive-streams-cpp/blob/master/include/ReactiveStreams.h